Add Node.js compatibility page to /platform section (#8320)
cloudflare/cloudflare-docs: content/workers/platform/nodejs-compatibility.md

---
pcx_content_type: concept

> title: Node.js Compatibility

Reviewer: This is a style guide format. You'll see this format in other titles and subtitles in dev platform docs.

---

> # Node.js Compatibility

Reviewer: ```suggestion
# Node.js compatibility
```

> The following APIs from the [Node.js runtime](https://nodejs.org/en/about) are available directly as [Runtime APIs](/workers/runtime-apis/nodejs), with no need to add polyfills to your own code:

Author: https://github.com/cloudflare/cloudflare-docs/pull/8320/commits/8aef609dcbaf405e59b3044982466e40b7266be9

> The following APIs from the [Node.js runtime](https://nodejs.org/en/about) are available directly as [Runtime APIs](/workers/runtime-apis/nodejs), with no need to add polyfills to your own code:

Author: @deadlypants1973 or @elithrar for 👀 on https://github.com/cloudflare/cloudflare-docs/commit/8aef609dcbaf405e59b3044982466e40b7266be9

{{<directory-listing folderDirectory="/workers/runtime-apis/nodejs/" >}}

> Node.js APIs are available under the `node:` prefix, and this prefix must be used when importing modules, both in your code and the NPM packages you depend on.

Reviewer 2: Is there an easy way to have dependencies reference these native built-ins?

Author: Not yet, but I know @jasnell has ideas

```js
// Do this:
import { Buffer } from 'node:buffer';

> // Not this:

Reviewer: ```suggestion
//  Do not do this:
```

import { Buffer } from 'buffer';
```

> ## How to enable with Workers

Reviewer: ```suggestion
## Enable Node.js with Workers
```

Add the [`nodejs_compat`](/workers/platform/compatibility-dates/#nodejs-compatibility-flag) [compatibility flag](/workers/platform/compatibility-dates/#nodejs-compatibility-flag) to your `wrangler.toml`:

```toml
---
header: wrangler.toml
---
compatibility_flags = [ "nodejs_compat" ]
```

<!-- Add once https://github.com/cloudflare/cloudflare-docs/pull/8322 is merged -->
<!-- {{<render file="_nodejs-compat-local-dev.md">}} -->

> ## How to enable with Pages Functions

Reviewer: ```suggestion
## Enable Node.js with Pages Functions
```

> ### Using Wrangler

Reviewer: ```suggestion
### Enable with Wrangler
```

To enable `nodejs_compat` in local development, pass the [`--compatibility-flags`](/workers/wrangler/commands/#dev-1) argument with the `nodejs_compat` flag to `wrangler pages dev`:

```sh
$ wrangler pages dev [<DIRECTORY>] --compatibility-flags="nodejs_compat" --experimental-local
```

For additional options, refer to the list of [Pages-specific CLI commands](/workers/wrangler/commands/#dev-1).

> ### Using the Cloudflare dashboard

Reviewer: ```suggestion
### Enable Node.js from the Cloudflare dashboard
```

> 

Reviewer: ```suggestion

To enable Node.js for your Pages Function from the Cloudflare dashboard:

```

> 1. Log into the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.

Reviewer: ```suggestion
1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.
```

2. Select **Pages** and select your Pages project.
3. Select **Settings** > **Functions** > **Compatibility Flags**.

> 4. Add the `nodejs_compat` compatibility flag to your Preview and Production deployments

Reviewer: ```suggestion
4. Add the `nodejs_compat` compatibility flag to your Preview and Production deployments.
```
